documentation/architecture/decisions/0003-crate-feature-flags.md

```diff
@@ -1,24 +1,62 @@
 # 3. Crate Feature Flags
 
-Date: 2021-02-10
+Date: 2021-09-20
 
 ## Status
 
 Proposed
 
 ## Context
 
-Ockam crates need guidelines for handling the following conditions in Rust
+Ockam wishes to be usable (and useful) in resource-constrained environments, including on embedded microcontrollers without a typical OS, and/or where dynamic memory allocation is not possible (either because resource limitations which make it impractical, or because of industrial standards which forbid it). As a result, our Rust crates must function in the following configurations:
 
-1. `no_std` Requires that memory be allocated statically at compile time. Some features might be enabled in this mode for stack based collection usage like `heapless`. This is stack only mode.
-1. `no_std+alloc` Standard collections provided by `alloc` can be used but they do not use the standard allocator. This allows the use of heap allocation via custom allocator(s).
-1. `std` normal Rust. This is standard mode.
+1. `no_std`: Requires that memory be allocated statically at compile time, and no global allocator is available. Some features might be enabled in this mode for stack based collection usage like `heapless`. This is stack only mode.
+1. `no_std+alloc`: Standard collections provided by `alloc` can be used but they do not use the standard allocator. This allows the use of heap allocation via custom allocator(s).
+1. `std`: normal Rust. This is the default mode.
 
-This document describes the various scenarios that can happen and how to address them. It also describes what is needed for end users of the system to understand what they get and ensure consistency across ockam crates.
+### Background: The Rust Standard Libraries
 
+The public API of Rust's standard library is split into 3 parts.
```

Comment:
This is really great background information to have here. Even if you have prior background it's not always clear what is what and why.

For instance, it hadn't ever occurred to me to think of `libstd` as a superset of both `libaloc` and `libcore`. For some reason I'd always categorized `liballoc` into a corner by itself.